### PR TITLE
Simpler help command

### PR DESCRIPTION
**Pull request by AndrewJWin**
Yay, we now have an embeded ``help`` command with categories.

### CODE REVIEW

File: commands/general/help.js [hunk 1]
```diff
@@ -1,55 +1,120 @@
...
     super(client, {
       name: "help",
-      description: "Displays the commands for your level.",
-      usage: "help [command]",
-      category: "System",
-      extended: "This command will display all available commands for your permission level, with the additonal option of getting per command information when you run 'help <command name>'.",
-      hidden: true,
-      aliases: ["h", "halp", "commands"]
+      description: "Get help on a command, command category, or a setting",
+      category: "Utility",
```
Comment (YorkAARGH) [inline commands/general/help.js]: This should be `System`

File: commands/general/help.js [hunk 2]
```diff
@@ -1,55 +1,120 @@
...
+      enabled:true,
```
Comment (YorkAARGH) [inline commands/general/help.js]: We don't have this.

File: commands/general/help.js [hunk 3]
```diff
@@ -1,55 +1,120 @@
...
+      guildOnly:false,
```
Comment (YorkAARGH) [inline commands/general/help.js]: Or this...

File: commands/general/help.js [hunk 4]
```diff
@@ -1,55 +1,120 @@
...
+      usage: "help <category/command/setting> [page-num]",
+      aliases: ["halp","h"],
+      permLevel: "User",
+      examples:["moderation","mute"]
```
Comment (YorkAARGH) [inline commands/general/help.js]: Or this...

File: commands/general/help.js [hunk 5]
```diff
@@ -1,55 +1,120 @@
...
-      extended: "This command will display all available commands for your permission level, with the additonal option of getting per command information when you run 'help <command name>'.",
-      hidden: true,
-      aliases: ["h", "halp", "commands"]
+      description: "Get help on a command, command category, or a setting",
+      category: "Utility",
+      enabled:true,
+      guildOnly:false,
+      usage: "help <category/command/setting> [page-num]",
+      aliases: ["halp","h"],
+      permLevel: "User",
```
Comment (YorkAARGH) [inline commands/general/help.js]: This isn't required as the `base` Command.js file does it for us.

File: commands/general/help.js [hunk 6]
```diff
@@ -1,55 +1,120 @@
 const Command = require(`${process.cwd()}/base/Command.js`);
-
+const {MessageEmbed} = require("discord.js");
```
Comment (YorkAARGH) [inline commands/general/help.js:2]: Pls change `{MessageEmbed}` to `{ MessageEmbed }`

File: commands/general/help.js [hunk 7]
```diff
@@ -1,55 +1,120 @@
...
     });
   }
 
   async run(message, args, level) {
-    const settings = message.settings;
+  // If no specific command is called, show all filtered commands.
+    const client = this.client;
+    const embed = new MessageEmbed();
+    // Load guild settings (for prefixes and eventually per-guild tweaks)
+    const settings = message.guild ? await this.client.settings.get(message.guild.id) : this.client.config.defaultSettings;
```
Comment (YorkAARGH) [inline commands/general/help.js]: Why remove the `settings` as we pass that via the [message event](https://github.com/NotAWeebDev/Misaki/blob/Yukine-Refactor/events/message.js#L13-L15), you're just duplicating code.

File: commands/general/help.js [hunk 8]
```diff
@@ -1,55 +1,120 @@
...
+      sorted.forEach( c => {
+        const cat = c.help.category.toProperCase();
+    
+        if (currentCategory !== cat && !args[0]) {
+          output += `\`${settings.prefix}help ${cat.toLowerCase()}\` | Shows ${cat} commands\n`;
+          currentCategory = cat;
+        }
+        //output += `${settings.prefix.value}${c.help.name}${" ".repeat(longest - c.help.name.length)} :: ${c.help.description}\n`;
+      });
+      embed.setTitle("Misaki Help")
```
Comment (YorkAARGH) [inline commands/general/help.js]: Why? We know it's the bot's help, pls remove the bot name

File: commands/general/help.js [hunk 9]
```diff
@@ -1,55 +1,120 @@
...
+        .setDescription(description)
+        .setAuthor(message.author.tag, message.author.avatarURL())
+        .addField("Categories", output)
+        .setTimestamp()
+        .setColor(message.guild.me.roles.highest.color || 0x00AE86)
+        .setFooter("Misaki", this.client.user.avatarURL());
```
Comment (YorkAARGH) [inline commands/general/help.js]: Change this to whom requested the help menu pls.

File: commands/general/help.js [hunk 10]
```diff
@@ -1,55 +1,120 @@
...
+    
+        if (currentCategory !== cat && !args[0]) {
+          output += `\`${settings.prefix}help ${cat.toLowerCase()}\` | Shows ${cat} commands\n`;
+          currentCategory = cat;
+        }
+        //output += `${settings.prefix.value}${c.help.name}${" ".repeat(longest - c.help.name.length)} :: ${c.help.description}\n`;
+      });
+      embed.setTitle("Misaki Help")
+        .setDescription(description)
+        .setAuthor(message.author.tag, message.author.avatarURL())
```
Comment (YorkAARGH) [inline commands/general/help.js]: Pls remove this.

File: commands/general/help.js [hunk 11]
```diff
@@ -1,55 +1,120 @@
...
+        }
+      });
+    
+    
+      if (num != 0) {
+        //message.channel.send(`${title}\n\n${description}\n${output}`, {code:"asciidoc"});
+        embed.setTitle("Command category help")
+          .setDescription(`A list of commands in the ${args[0]} category.  (Total of ${num} commands in this category)\n\nTo get help on a specific command do \`${settings.prefix}help <command>\``)
+          .addField("Commands", output)
+          .setAuthor(message.author.tag, message.author.avatarURL())
```
Comment (YorkAARGH) [inline commands/general/help.js]: Pls change this to the footer, to be more in tune with the rest of the commands

File: commands/general/help.js [hunk 12]
```diff
@@ -1,55 +1,120 @@
...
-      message.channel.send(`= ${command.help.name} = \n${command.help.description}\ncategory     :: ${command.help.category}\ncost         :: ₲${parseInt(command.help.cost)}\nusage        :: ${command.help.usage}\naliases      :: ${command.conf.aliases.join(", ")}\ndetails      :: ${command.help.extended}`, {code:"asciidoc"});    }
+    }
+
+    // Show individual command's help.
+    let command = args[0];
+    if (client.commands.has(command) || client.commands.forEach(command => {if (command.conf.aliases.includes(command)) return true;})) {
+      command = client.commands.get(command);
+      if (level < client.levelCache[command.conf.permLevel]) return;
+      embed.setTitle(`${args[0]} help`)
+        .setAuthor(message.author.tag, message.author.avatarURL())
```
Comment (YorkAARGH) [inline commands/general/help.js]: Again, move this to the footer.

**Review by YorkAARGH — CHANGES_REQUESTED** [review]
Overall this is going to help Misaki a lot, but it's certainly a "drop in" file from another bot as it has properties that Misaki doesn't have.

**Comment by AndrewJWin** [comment]
More changes are needed, will close and reopen when ready.

**Pull request closed without merging**